### ISSUE TITLE
Improve (consistency of) rendering of (cycle/hiking/etc) routes

### ISSUE THREAD

**Issue by matthijskooijman** [osmandapp/OsmAnd#9597]
Part of this was previously commented at https://github.com/osmandapp/OsmAnd/issues/9195#issuecomment-657454728, but I think this warrants an issue of its own, so here it is.

It seems that there are inconsistencies between the different route renderings:

 - Hiking routes with "Color by OSMC hiking symbol" are quite transparent and the route is rendered below the road.
 - Hiking routes with "Color by network affiliation" are quite opaque and renderd on top of the road.
 - Cycle routes are quite transparent and rendered on top of the road.

I'm not sure if there is good reason for this inconsistency, or it is just a matter of selectively applied improvements, but maybe it would be good to use more consistent and easier to use route rendering. Especially if the rendering can be additionally customized (which I think is the subject of #9195), improving the defaults should not be too problematic.

### Evaluating current rendering styles

When you're interested in the routes, the "Color by network affiliation" using nearly opaque route lines makes the routes easiest to distinguish. However, this also makes it very hard or impossible to see the road type (or surface/quality when enabled), especially when the route is rendered wider than the road.

The transparent on-top cycle routes better allow viewing the road type and surface details, but this is still quite hard to see without extra zooming in. Additionally, transparent routes on top of the road means that the route changes color slightly depending on the color of the road, which makes it harder to distinguish different route types (e.g. node vs national vs regional etc.).

The transparent below-the-road style used by "Color by OSMC symbol" makes it easiest to still distinguish road types and quality. It does still suffer from variations in the route coloring due to transparency, e.g. comparing a route going through a forest (with a green background) or through a residential area (grey background). The transparency sometimes hampers visibility too. 

I believe this last style would be a good starting point, though maybe with a bit more opaque rendering.

### Zoom-level dependent rendering
A related improvement could be to make the opacity dependent on the zoomlevel. When you are zoomed out, you probably want an overview of the route (which calls for higher opacity and maybe rendering on top of the road with a narrower route line), but when you are zoomed in, you more likely want to see road details (which calls for higher transparency and/or rendering below the road). Though even when zoomed out, it might still be relevant to distinguish primary/secondary highways from other highways, so completely hiding that might not be ideal either.

### Ideal default

I'm not quite sure what the ideal default rendering would look like yet, this probably requires some experimentation. I'll give this a go when I find some time, but already wanted to share some thoughts on this first.

**Comment by vshcherb** [comment]
This one is quite difficult to understand how to achieve. I think the option names are not clear for end user and we will need to  change them to have better explanation, so it will be a subject for change in the next release.

**Comment by matthijskooijman** [comment]
Yeah, I'm not sure what the best approach would be here either, just that there is some room for improvement :-)

**Comment by snouup** [comment]
Transparent rendering in all cases would be great. As of now, the opaque Color by network affiliation mode requires zooming in all the way, or even disable hiking symbols completely, just to see the type of path underneath. I would like to see the type of path to judge the quality of a hiking trail (as in percentage of paved roads). I can't see the roads with the opaque overlay on top.
